// @fluentui/react-popover/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Popover: function() {
        return _Popover.Popover;
    },
    PopoverProvider: function() {
        return _popoverContext.PopoverProvider;
    },
    PopoverSurface: function() {
        return _PopoverSurface.PopoverSurface;
    },
    PopoverTrigger: function() {
        return _PopoverTrigger.PopoverTrigger;
    },
    arrowHeights: function() {
        return _PopoverSurface.arrowHeights;
    },
    popoverSurfaceClassNames: function() {
        return _PopoverSurface.popoverSurfaceClassNames;
    },
    renderPopoverSurface_unstable: function() {
        return _PopoverSurface.renderPopoverSurface_unstable;
    },
    renderPopoverTrigger_unstable: function() {
        return _PopoverTrigger.renderPopoverTrigger_unstable;
    },
    renderPopover_unstable: function() {
        return _Popover.renderPopover_unstable;
    },
    usePopoverContext_unstable: function() {
        return _popoverContext.usePopoverContext_unstable;
    },
    usePopoverSurfaceStyles_unstable: function() {
        return _PopoverSurface.usePopoverSurfaceStyles_unstable;
    },
    usePopoverSurface_unstable: function() {
        return _PopoverSurface.usePopoverSurface_unstable;
    },
    usePopoverTrigger_unstable: function() {
        return _PopoverTrigger.usePopoverTrigger_unstable;
    },
    usePopover_unstable: function() {
        return _Popover.usePopover_unstable;
    }
});
const _Popover = require("./Popover");
const _PopoverSurface = require("./PopoverSurface");
const _popoverContext = require("./popoverContext");
const _PopoverTrigger = require("./PopoverTrigger");
