// @fluentui/react-popover/lib-commonjs/components/PopoverSurface/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    PopoverSurface: function() {
        return _PopoverSurface.PopoverSurface;
    },
    arrowHeights: function() {
        return _usePopoverSurfaceStylesstyles.arrowHeights;
    },
    popoverSurfaceClassNames: function() {
        return _usePopoverSurfaceStylesstyles.popoverSurfaceClassNames;
    },
    renderPopoverSurface_unstable: function() {
        return _renderPopoverSurface.renderPopoverSurface_unstable;
    },
    usePopoverSurfaceStyles_unstable: function() {
        return _usePopoverSurfaceStylesstyles.usePopoverSurfaceStyles_unstable;
    },
    usePopoverSurface_unstable: function() {
        return _usePopoverSurface.usePopoverSurface_unstable;
    }
});
const _PopoverSurface = require("./PopoverSurface");
const _renderPopoverSurface = require("./renderPopoverSurface");
const _usePopoverSurface = require("./usePopoverSurface");
const _usePopoverSurfaceStylesstyles = require("./usePopoverSurfaceStyles.styles");
