// @fluentui/react-link/lib-commonjs/components/Link/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Link: function() {
        return _Link.Link;
    },
    linkClassNames: function() {
        return _useLinkStylesstyles.linkClassNames;
    },
    renderLink_unstable: function() {
        return _renderLink.renderLink_unstable;
    },
    useLinkState_unstable: function() {
        return _useLinkState.useLinkState_unstable;
    },
    useLinkStyles_unstable: function() {
        return _useLinkStylesstyles.useLinkStyles_unstable;
    },
    useLink_unstable: function() {
        return _useLink.useLink_unstable;
    }
});
const _Link = require("./Link");
const _renderLink = require("./renderLink");
const _useLink = require("./useLink");
const _useLinkState = require("./useLinkState");
const _useLinkStylesstyles = require("./useLinkStyles.styles");
