// @fluentui/react-card/lib-commonjs/Card.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Card: function() {
        return _index.Card;
    },
    CardProvider: function() {
        return _index.CardProvider;
    },
    cardCSSVars: function() {
        return _index.cardCSSVars;
    },
    cardClassNames: function() {
        return _index.cardClassNames;
    },
    cardContextDefaultValue: function() {
        return _index.cardContextDefaultValue;
    },
    renderCard_unstable: function() {
        return _index.renderCard_unstable;
    },
    useCardContext_unstable: function() {
        return _index.useCardContext_unstable;
    },
    useCardStyles_unstable: function() {
        return _index.useCardStyles_unstable;
    },
    useCard_unstable: function() {
        return _index.useCard_unstable;
    }
});
const _index = require("./components/Card/index");
