// @fluentui/react-provider/lib-commonjs/components/FluentProvider/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FluentProvider: function() {
        return _FluentProvider.FluentProvider;
    },
    createCSSRuleFromTheme: function() {
        return _createCSSRuleFromTheme.createCSSRuleFromTheme;
    },
    fluentProviderClassNames: function() {
        return _useFluentProviderStylesstyles.fluentProviderClassNames;
    },
    renderFluentProvider_unstable: function() {
        return _renderFluentProvider.renderFluentProvider_unstable;
    },
    useFluentProviderContextValues_unstable: function() {
        return _useFluentProviderContextValues.useFluentProviderContextValues_unstable;
    },
    useFluentProviderStyles_unstable: function() {
        return _useFluentProviderStylesstyles.useFluentProviderStyles_unstable;
    },
    useFluentProviderThemeStyleTag: function() {
        return _useFluentProviderThemeStyleTag.useFluentProviderThemeStyleTag;
    },
    useFluentProvider_unstable: function() {
        return _useFluentProvider.useFluentProvider_unstable;
    }
});
const _createCSSRuleFromTheme = require("./createCSSRuleFromTheme");
const _FluentProvider = require("./FluentProvider");
const _renderFluentProvider = require("./renderFluentProvider");
const _useFluentProvider = require("./useFluentProvider");
const _useFluentProviderStylesstyles = require("./useFluentProviderStyles.styles");
const _useFluentProviderContextValues = require("./useFluentProviderContextValues");
const _useFluentProviderThemeStyleTag = require("./useFluentProviderThemeStyleTag");
