// @fluentui/react-tags/lib-commonjs/components/InteractionTag/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    InteractionTag: function() {
        return _InteractionTag.InteractionTag;
    },
    interactionTagClassNames: function() {
        return _useInteractionTagStylesstyles.interactionTagClassNames;
    },
    renderInteractionTag_unstable: function() {
        return _renderInteractionTag.renderInteractionTag_unstable;
    },
    useInteractionTagContextValues_unstable: function() {
        return _useInteractionTagContextValues.useInteractionTagContextValues_unstable;
    },
    useInteractionTagStyles_unstable: function() {
        return _useInteractionTagStylesstyles.useInteractionTagStyles_unstable;
    },
    useInteractionTag_unstable: function() {
        return _useInteractionTag.useInteractionTag_unstable;
    }
});
const _InteractionTag = require("./InteractionTag");
const _renderInteractionTag = require("./renderInteractionTag");
const _useInteractionTag = require("./useInteractionTag");
const _useInteractionTagStylesstyles = require("./useInteractionTagStyles.styles");
const _useInteractionTagContextValues = require("./useInteractionTagContextValues");
