// @fluentui/react-table/lib-commonjs/components/TableRow/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TableRow: function() {
        return _TableRow.TableRow;
    },
    renderTableRow_unstable: function() {
        return _renderTableRow.renderTableRow_unstable;
    },
    tableRowClassName: function() {
        return _useTableRowStylesstyles.tableRowClassName;
    },
    tableRowClassNames: function() {
        return _useTableRowStylesstyles.tableRowClassNames;
    },
    useTableRowStyles_unstable: function() {
        return _useTableRowStylesstyles.useTableRowStyles_unstable;
    },
    useTableRow_unstable: function() {
        return _useTableRow.useTableRow_unstable;
    }
});
const _TableRow = require("./TableRow");
const _renderTableRow = require("./renderTableRow");
const _useTableRow = require("./useTableRow");
const _useTableRowStylesstyles = require("./useTableRowStyles.styles");
