// @fluentui/react-message-bar/lib-commonjs/contexts/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MessageBarContextProvider: function() {
        return _messageBarContext.MessageBarContextProvider;
    },
    MessageBarTransitionContextProvider: function() {
        return _messageBarTransitionContext.MessageBarTransitionContextProvider;
    },
    messageBarContextDefaultValue: function() {
        return _messageBarContext.messageBarContextDefaultValue;
    },
    messageBarTransitionContextDefaultValue: function() {
        return _messageBarTransitionContext.messageBarTransitionContextDefaultValue;
    },
    useMessageBarContext: function() {
        return _messageBarContext.useMessageBarContext;
    },
    useMessageBarTransitionContext: function() {
        return _messageBarTransitionContext.useMessageBarTransitionContext;
    }
});
const _messageBarContext = require("./messageBarContext");
const _messageBarTransitionContext = require("./messageBarTransitionContext");
