// @fluentui/react-rating/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Rating: function() {
        return _Rating.Rating;
    },
    RatingDisplay: function() {
        return _RatingDisplay.RatingDisplay;
    },
    RatingItem: function() {
        return _RatingItem.RatingItem;
    },
    RatingItemProvider: function() {
        return _index.RatingItemProvider;
    },
    ratingClassNames: function() {
        return _Rating.ratingClassNames;
    },
    ratingDisplayClassNames: function() {
        return _RatingDisplay.ratingDisplayClassNames;
    },
    ratingItemClassNames: function() {
        return _RatingItem.ratingItemClassNames;
    },
    renderRatingDisplay_unstable: function() {
        return _RatingDisplay.renderRatingDisplay_unstable;
    },
    renderRatingItem_unstable: function() {
        return _RatingItem.renderRatingItem_unstable;
    },
    renderRating_unstable: function() {
        return _Rating.renderRating_unstable;
    },
    useRatingContextValues: function() {
        return _Rating.useRatingContextValues;
    },
    useRatingDisplayContextValues: function() {
        return _RatingDisplay.useRatingDisplayContextValues;
    },
    useRatingDisplayStyles_unstable: function() {
        return _RatingDisplay.useRatingDisplayStyles_unstable;
    },
    useRatingDisplay_unstable: function() {
        return _RatingDisplay.useRatingDisplay_unstable;
    },
    useRatingItemContextValue_unstable: function() {
        return _index.useRatingItemContextValue_unstable;
    },
    useRatingItemStyles_unstable: function() {
        return _RatingItem.useRatingItemStyles_unstable;
    },
    useRatingItem_unstable: function() {
        return _RatingItem.useRatingItem_unstable;
    },
    useRatingStyles_unstable: function() {
        return _Rating.useRatingStyles_unstable;
    },
    useRating_unstable: function() {
        return _Rating.useRating_unstable;
    }
});
const _Rating = require("./Rating");
const _RatingItem = require("./RatingItem");
const _index = require("./contexts/index");
const _RatingDisplay = require("./RatingDisplay");
