// @fluentui/react-accordion/lib-commonjs/components/AccordionHeader/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AccordionHeader: function() {
        return _AccordionHeader.AccordionHeader;
    },
    accordionHeaderClassNames: function() {
        return _useAccordionHeaderStylesstyles.accordionHeaderClassNames;
    },
    renderAccordionHeader_unstable: function() {
        return _renderAccordionHeader.renderAccordionHeader_unstable;
    },
    useAccordionHeaderContextValues_unstable: function() {
        return _useAccordionHeaderContextValues.useAccordionHeaderContextValues_unstable;
    },
    useAccordionHeaderStyles_unstable: function() {
        return _useAccordionHeaderStylesstyles.useAccordionHeaderStyles_unstable;
    },
    useAccordionHeader_unstable: function() {
        return _useAccordionHeader.useAccordionHeader_unstable;
    }
});
const _AccordionHeader = require("./AccordionHeader");
const _renderAccordionHeader = require("./renderAccordionHeader");
const _useAccordionHeader = require("./useAccordionHeader");
const _useAccordionHeaderContextValues = require("./useAccordionHeaderContextValues");
const _useAccordionHeaderStylesstyles = require("./useAccordionHeaderStyles.styles");
