// @fluentui/react-slider/lib-commonjs/components/Slider/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Slider: function() {
        return _Slider.Slider;
    },
    renderSlider_unstable: function() {
        return _renderSlider.renderSlider_unstable;
    },
    sliderCSSVars: function() {
        return _useSliderStylesstyles.sliderCSSVars;
    },
    sliderClassNames: function() {
        return _useSliderStylesstyles.sliderClassNames;
    },
    useSliderState_unstable: function() {
        return _useSliderState.useSliderState_unstable;
    },
    useSliderStyles_unstable: function() {
        return _useSliderStylesstyles.useSliderStyles_unstable;
    },
    useSlider_unstable: function() {
        return _useSlider.useSlider_unstable;
    }
});
const _Slider = require("./Slider");
const _renderSlider = require("./renderSlider");
const _useSlider = require("./useSlider");
const _useSliderState = require("./useSliderState");
const _useSliderStylesstyles = require("./useSliderStyles.styles");
