// @fluentui/react-swatch-picker/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ColorSwatch: function() {
        return _ColorSwatch.ColorSwatch;
    },
    EmptySwatch: function() {
        return _EmptySwatch.EmptySwatch;
    },
    ImageSwatch: function() {
        return _ImageSwatch.ImageSwatch;
    },
    SwatchPicker: function() {
        return _SwatchPicker.SwatchPicker;
    },
    SwatchPickerProvider: function() {
        return _contexts.SwatchPickerProvider;
    },
    SwatchPickerRow: function() {
        return _SwatchPickerRow.SwatchPickerRow;
    },
    colorSwatchClassNames: function() {
        return _ColorSwatch.colorSwatchClassNames;
    },
    emptySwatchClassNames: function() {
        return _EmptySwatch.emptySwatchClassNames;
    },
    imageSwatchClassNames: function() {
        return _ImageSwatch.imageSwatchClassNames;
    },
    renderColorSwatch_unstable: function() {
        return _ColorSwatch.renderColorSwatch_unstable;
    },
    renderEmptySwatch_unstable: function() {
        return _EmptySwatch.renderEmptySwatch_unstable;
    },
    renderImageSwatch_unstable: function() {
        return _ImageSwatch.renderImageSwatch_unstable;
    },
    renderSwatchPickerGrid: function() {
        return _renderUtils.renderSwatchPickerGrid;
    },
    renderSwatchPickerRow_unstable: function() {
        return _SwatchPickerRow.renderSwatchPickerRow_unstable;
    },
    renderSwatchPicker_unstable: function() {
        return _SwatchPicker.renderSwatchPicker_unstable;
    },
    swatchCSSVars: function() {
        return _ColorSwatch.swatchCSSVars;
    },
    swatchPickerClassNames: function() {
        return _SwatchPicker.swatchPickerClassNames;
    },
    swatchPickerContextDefaultValue: function() {
        return _contexts.swatchPickerContextDefaultValue;
    },
    swatchPickerRowClassNames: function() {
        return _SwatchPickerRow.swatchPickerRowClassNames;
    },
    useColorSwatchStyles_unstable: function() {
        return _ColorSwatch.useColorSwatchStyles_unstable;
    },
    useColorSwatch_unstable: function() {
        return _ColorSwatch.useColorSwatch_unstable;
    },
    useEmptySwatchStyles_unstable: function() {
        return _EmptySwatch.useEmptySwatchStyles_unstable;
    },
    useEmptySwatch_unstable: function() {
        return _EmptySwatch.useEmptySwatch_unstable;
    },
    useImageSwatchStyles_unstable: function() {
        return _ImageSwatch.useImageSwatchStyles_unstable;
    },
    useImageSwatch_unstable: function() {
        return _ImageSwatch.useImageSwatch_unstable;
    },
    useSwatchPickerContextValue_unstable: function() {
        return _contexts.useSwatchPickerContextValue_unstable;
    },
    useSwatchPickerContextValues: function() {
        return _contexts.useSwatchPickerContextValues;
    },
    useSwatchPickerRowStyles_unstable: function() {
        return _SwatchPickerRow.useSwatchPickerRowStyles_unstable;
    },
    useSwatchPickerRow_unstable: function() {
        return _SwatchPickerRow.useSwatchPickerRow_unstable;
    },
    useSwatchPickerStyles_unstable: function() {
        return _SwatchPicker.useSwatchPickerStyles_unstable;
    },
    useSwatchPicker_unstable: function() {
        return _SwatchPicker.useSwatchPicker_unstable;
    }
});
const _SwatchPicker = require("./SwatchPicker");
const _contexts = require("./contexts");
const _ColorSwatch = require("./ColorSwatch");
const _ImageSwatch = require("./ImageSwatch");
const _SwatchPickerRow = require("./SwatchPickerRow");
const _renderUtils = require("./utils/renderUtils");
const _EmptySwatch = require("./EmptySwatch");
