// @fluentui/react-virtualizer/lib-commonjs/hooks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    createResizeObserverFromDocument: function() {
        return _useMeasureList.createResizeObserverFromDocument;
    },
    getRTLRootMargin: function() {
        return _useIntersectionObserver.getRTLRootMargin;
    },
    useDynamicVirtualizerMeasure: function() {
        return _useDynamicVirtualizerMeasure.useDynamicVirtualizerMeasure;
    },
    useIntersectionObserver: function() {
        return _useIntersectionObserver.useIntersectionObserver;
    },
    useMeasureList: function() {
        return _useMeasureList.useMeasureList;
    },
    useResizeObserverRef_unstable: function() {
        return _useResizeObserverRef.useResizeObserverRef_unstable;
    },
    useStaticVirtualizerMeasure: function() {
        return _useVirtualizerMeasure.useStaticVirtualizerMeasure;
    }
});
const _useIntersectionObserver = require("./useIntersectionObserver");
const _useVirtualizerMeasure = require("./useVirtualizerMeasure");
const _useDynamicVirtualizerMeasure = require("./useDynamicVirtualizerMeasure");
const _useResizeObserverRef = require("./useResizeObserverRef");
const _useMeasureList = require("./useMeasureList");
