// @fluentui/react-swatch-picker/lib-commonjs/components/ColorSwatch/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ColorSwatch: function() {
        return _ColorSwatch.ColorSwatch;
    },
    colorSwatchClassNames: function() {
        return _useColorSwatchStylesstyles.colorSwatchClassNames;
    },
    renderColorSwatch_unstable: function() {
        return _renderColorSwatch.renderColorSwatch_unstable;
    },
    swatchCSSVars: function() {
        return _useColorSwatchStylesstyles.swatchCSSVars;
    },
    useColorSwatchStyles_unstable: function() {
        return _useColorSwatchStylesstyles.useColorSwatchStyles_unstable;
    },
    useColorSwatch_unstable: function() {
        return _useColorSwatch.useColorSwatch_unstable;
    }
});
const _ColorSwatch = require("./ColorSwatch");
const _renderColorSwatch = require("./renderColorSwatch");
const _useColorSwatch = require("./useColorSwatch");
const _useColorSwatchStylesstyles = require("./useColorSwatchStyles.styles");
