// @fluentui/react-toast/lib-commonjs/state/vanilla/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    createToaster: function() {
        return _createToaster.createToaster;
    },
    dismissAllToasts: function() {
        return _dismissAllToasts.dismissAllToasts;
    },
    dismissToast: function() {
        return _dismissToast.dismissToast;
    },
    dispatchToast: function() {
        return _dispatchToast.dispatchToast;
    },
    getPositionStyles: function() {
        return _getPositionStyles.getPositionStyles;
    },
    pauseToast: function() {
        return _pauseToast.pauseToast;
    },
    playToast: function() {
        return _playToast.playToast;
    },
    updateToast: function() {
        return _updateToast.updateToast;
    }
});
const _dispatchToast = require("./dispatchToast");
const _dismissToast = require("./dismissToast");
const _dismissAllToasts = require("./dismissAllToasts");
const _updateToast = require("./updateToast");
const _pauseToast = require("./pauseToast");
const _playToast = require("./playToast");
const _createToaster = require("./createToaster");
const _getPositionStyles = require("./getPositionStyles");
