// @fluentui/react-dialog/lib-commonjs/components/DialogSurface/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DialogSurface: function() {
        return _DialogSurface.DialogSurface;
    },
    dialogSurfaceClassNames: function() {
        return _useDialogSurfaceStylesstyles.dialogSurfaceClassNames;
    },
    renderDialogSurface_unstable: function() {
        return _renderDialogSurface.renderDialogSurface_unstable;
    },
    useDialogSurfaceContextValues_unstable: function() {
        return _useDialogSurfaceContextValues.useDialogSurfaceContextValues_unstable;
    },
    useDialogSurfaceStyles_unstable: function() {
        return _useDialogSurfaceStylesstyles.useDialogSurfaceStyles_unstable;
    },
    useDialogSurface_unstable: function() {
        return _useDialogSurface.useDialogSurface_unstable;
    }
});
const _DialogSurface = require("./DialogSurface");
const _renderDialogSurface = require("./renderDialogSurface");
const _useDialogSurface = require("./useDialogSurface");
const _useDialogSurfaceStylesstyles = require("./useDialogSurfaceStyles.styles");
const _useDialogSurfaceContextValues = require("./useDialogSurfaceContextValues");
