// @fluentui/react-utilities/lib-commonjs/hooks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    IdPrefixProvider: function() {
        return _useId.IdPrefixProvider;
    },
    resetIdsForTests: function() {
        return _useId.resetIdsForTests;
    },
    useAnimationFrame: function() {
        return _useAnimationFrame.useAnimationFrame;
    },
    useControllableState: function() {
        return _useControllableState.useControllableState;
    },
    useEventCallback: function() {
        return _useEventCallback.useEventCallback;
    },
    useFirstMount: function() {
        return _useFirstMount.useFirstMount;
    },
    useForceUpdate: function() {
        return _useForceUpdate.useForceUpdate;
    },
    useId: function() {
        return _useId.useId;
    },
    useIsomorphicLayoutEffect: function() {
        return _useIsomorphicLayoutEffect.useIsomorphicLayoutEffect;
    },
    useMergedRefs: function() {
        return _useMergedRefs.useMergedRefs;
    },
    useOnClickOutside: function() {
        return _useOnClickOutside.useOnClickOutside;
    },
    useOnScrollOutside: function() {
        return _useOnScrollOutside.useOnScrollOutside;
    },
    usePrevious: function() {
        return _usePrevious.usePrevious;
    },
    useScrollbarWidth: function() {
        return _useScrollbarWidth.useScrollbarWidth;
    },
    useTimeout: function() {
        return _useTimeout.useTimeout;
    }
});
const _useAnimationFrame = require("./useAnimationFrame");
const _useControllableState = require("./useControllableState");
const _useEventCallback = require("./useEventCallback");
const _useFirstMount = require("./useFirstMount");
const _useForceUpdate = require("./useForceUpdate");
const _useId = require("./useId");
const _useIsomorphicLayoutEffect = require("./useIsomorphicLayoutEffect");
const _useMergedRefs = require("./useMergedRefs");
const _useOnClickOutside = require("./useOnClickOutside");
const _useOnScrollOutside = require("./useOnScrollOutside");
const _usePrevious = require("./usePrevious");
const _useScrollbarWidth = require("./useScrollbarWidth");
const _useTimeout = require("./useTimeout");
