// @fluentui/react-positioning/lib-commonjs/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    debounce: function() {
        return _debounce.debounce;
    },
    fromFloatingUIPlacement: function() {
        return _fromFloatingUIPlacement.fromFloatingUIPlacement;
    },
    getBoundary: function() {
        return _getBoundary.getBoundary;
    },
    getParentNode: function() {
        return _getScrollParent.getParentNode;
    },
    getReactFiberFromNode: function() {
        return _getReactFiberFromNode.getReactFiberFromNode;
    },
    getScrollParent: function() {
        return _getScrollParent.getScrollParent;
    },
    hasAutofocusFilter: function() {
        return _hasAutoFocusFilter.hasAutofocusFilter;
    },
    hasScrollParent: function() {
        return _getScrollParent.hasScrollParent;
    },
    mergeArrowOffset: function() {
        return _mergeArrowOffset.mergeArrowOffset;
    },
    normalizeAutoSize: function() {
        return _normalizeAutoSize.normalizeAutoSize;
    },
    parseFloatingUIPlacement: function() {
        return _parseFloatingUIPlacement.parseFloatingUIPlacement;
    },
    resolvePositioningShorthand: function() {
        return _resolvePositioningShorthand.resolvePositioningShorthand;
    },
    toFloatingUIPadding: function() {
        return _toFloatingUIPadding.toFloatingUIPadding;
    },
    toFloatingUIPlacement: function() {
        return _toFloatingUIPlacement.toFloatingUIPlacement;
    },
    toggleScrollListener: function() {
        return _toggleScrollListener.toggleScrollListener;
    },
    useCallbackRef: function() {
        return _useCallbackRef.useCallbackRef;
    },
    writeArrowUpdates: function() {
        return _writeArrowUpdates.writeArrowUpdates;
    },
    writeContainerUpdates: function() {
        return _writeContainerupdates.writeContainerUpdates;
    }
});
const _parseFloatingUIPlacement = require("./parseFloatingUIPlacement");
const _getBoundary = require("./getBoundary");
const _getReactFiberFromNode = require("./getReactFiberFromNode");
const _getScrollParent = require("./getScrollParent");
const _mergeArrowOffset = require("./mergeArrowOffset");
const _toFloatingUIPadding = require("./toFloatingUIPadding");
const _toFloatingUIPlacement = require("./toFloatingUIPlacement");
const _fromFloatingUIPlacement = require("./fromFloatingUIPlacement");
const _resolvePositioningShorthand = require("./resolvePositioningShorthand");
const _useCallbackRef = require("./useCallbackRef");
const _debounce = require("./debounce");
const _toggleScrollListener = require("./toggleScrollListener");
const _hasAutoFocusFilter = require("./hasAutoFocusFilter");
const _writeArrowUpdates = require("./writeArrowUpdates");
const _writeContainerupdates = require("./writeContainerupdates");
const _normalizeAutoSize = require("./normalizeAutoSize");
