// @fluentui/react-tags/lib-commonjs/components/TagGroup/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TagGroup: function() {
        return _TagGroup.TagGroup;
    },
    renderTagGroup_unstable: function() {
        return _renderTagGroup.renderTagGroup_unstable;
    },
    tagGroupClassNames: function() {
        return _useTagGroupStylesstyles.tagGroupClassNames;
    },
    useTagGroupContextValues_unstable: function() {
        return _useTagGroupContextValues.useTagGroupContextValues_unstable;
    },
    useTagGroupStyles_unstable: function() {
        return _useTagGroupStylesstyles.useTagGroupStyles_unstable;
    },
    useTagGroup_unstable: function() {
        return _useTagGroup.useTagGroup_unstable;
    }
});
const _TagGroup = require("./TagGroup");
const _renderTagGroup = require("./renderTagGroup");
const _useTagGroup = require("./useTagGroup");
const _useTagGroupStylesstyles = require("./useTagGroupStyles.styles");
const _useTagGroupContextValues = require("./useTagGroupContextValues");
