// @fluentui/react-card/lib-commonjs/components/CardHeader/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    CardHeader: function() {
        return _CardHeader.CardHeader;
    },
    cardHeaderCSSVars: function() {
        return _useCardHeaderStylesstyles.cardHeaderCSSVars;
    },
    cardHeaderClassNames: function() {
        return _useCardHeaderStylesstyles.cardHeaderClassNames;
    },
    renderCardHeader_unstable: function() {
        return _renderCardHeader.renderCardHeader_unstable;
    },
    useCardHeaderStyles_unstable: function() {
        return _useCardHeaderStylesstyles.useCardHeaderStyles_unstable;
    },
    useCardHeader_unstable: function() {
        return _useCardHeader.useCardHeader_unstable;
    }
});
const _CardHeader = require("./CardHeader");
const _renderCardHeader = require("./renderCardHeader");
const _useCardHeader = require("./useCardHeader");
const _useCardHeaderStylesstyles = require("./useCardHeaderStyles.styles");
