// @fluentui/react-table/lib-commonjs/components/TableCell/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TableCell: function() {
        return _TableCell.TableCell;
    },
    renderTableCell_unstable: function() {
        return _renderTableCell.renderTableCell_unstable;
    },
    tableCellClassName: function() {
        return _useTableCellStylesstyles.tableCellClassName;
    },
    tableCellClassNames: function() {
        return _useTableCellStylesstyles.tableCellClassNames;
    },
    useTableCellStyles_unstable: function() {
        return _useTableCellStylesstyles.useTableCellStyles_unstable;
    },
    useTableCell_unstable: function() {
        return _useTableCell.useTableCell_unstable;
    }
});
const _TableCell = require("./TableCell");
const _renderTableCell = require("./renderTableCell");
const _useTableCell = require("./useTableCell");
const _useTableCellStylesstyles = require("./useTableCellStyles.styles");
