// @fluentui/react-table/lib-commonjs/components/TableHeader/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TableHeader: function() {
        return _TableHeader.TableHeader;
    },
    renderTableHeader_unstable: function() {
        return _renderTableHeader.renderTableHeader_unstable;
    },
    tableHeaderClassName: function() {
        return _useTableHeaderStylesstyles.tableHeaderClassName;
    },
    tableHeaderClassNames: function() {
        return _useTableHeaderStylesstyles.tableHeaderClassNames;
    },
    useTableHeaderStyles_unstable: function() {
        return _useTableHeaderStylesstyles.useTableHeaderStyles_unstable;
    },
    useTableHeader_unstable: function() {
        return _useTableHeader.useTableHeader_unstable;
    }
});
const _TableHeader = require("./TableHeader");
const _renderTableHeader = require("./renderTableHeader");
const _useTableHeader = require("./useTableHeader");
const _useTableHeaderStylesstyles = require("./useTableHeaderStyles.styles");
