// @fluentui/react-breadcrumb/lib-commonjs/Breadcrumb.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Breadcrumb: function() {
        return _index.Breadcrumb;
    },
    BreadcrumbProvider: function() {
        return _index.BreadcrumbProvider;
    },
    breadcrumbClassNames: function() {
        return _index.breadcrumbClassNames;
    },
    breadcrumbDefaultValue: function() {
        return _index.breadcrumbDefaultValue;
    },
    renderBreadcrumb_unstable: function() {
        return _index.renderBreadcrumb_unstable;
    },
    useBreadcrumbContext_unstable: function() {
        return _index.useBreadcrumbContext_unstable;
    },
    useBreadcrumbStyles_unstable: function() {
        return _index.useBreadcrumbStyles_unstable;
    },
    useBreadcrumb_unstable: function() {
        return _index.useBreadcrumb_unstable;
    }
});
const _index = require("./components/Breadcrumb/index");
