// @fluentui/react-breadcrumb/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Breadcrumb: function() {
        return _Breadcrumb.Breadcrumb;
    },
    BreadcrumbButton: function() {
        return _BreadcrumbButton.BreadcrumbButton;
    },
    BreadcrumbDivider: function() {
        return _BreadcrumbDivider.BreadcrumbDivider;
    },
    BreadcrumbItem: function() {
        return _BreadcrumbItem.BreadcrumbItem;
    },
    BreadcrumbProvider: function() {
        return _Breadcrumb.BreadcrumbProvider;
    },
    breadcrumbButtonClassNames: function() {
        return _BreadcrumbButton.breadcrumbButtonClassNames;
    },
    breadcrumbClassNames: function() {
        return _Breadcrumb.breadcrumbClassNames;
    },
    breadcrumbDividerClassNames: function() {
        return _BreadcrumbDivider.breadcrumbDividerClassNames;
    },
    breadcrumbItemClassNames: function() {
        return _BreadcrumbItem.breadcrumbItemClassNames;
    },
    isTruncatableBreadcrumbContent: function() {
        return _index.isTruncatableBreadcrumbContent;
    },
    partitionBreadcrumbItems: function() {
        return _index.partitionBreadcrumbItems;
    },
    renderBreadcrumbButton_unstable: function() {
        return _BreadcrumbButton.renderBreadcrumbButton_unstable;
    },
    renderBreadcrumbDivider_unstable: function() {
        return _BreadcrumbDivider.renderBreadcrumbDivider_unstable;
    },
    renderBreadcrumbItem_unstable: function() {
        return _BreadcrumbItem.renderBreadcrumbItem_unstable;
    },
    renderBreadcrumb_unstable: function() {
        return _Breadcrumb.renderBreadcrumb_unstable;
    },
    truncateBreadcrumLongTooltip: function() {
        return _index.truncateBreadcrumLongTooltip;
    },
    truncateBreadcrumbLongName: function() {
        return _index.truncateBreadcrumbLongName;
    },
    useBreadcrumbButtonStyles_unstable: function() {
        return _BreadcrumbButton.useBreadcrumbButtonStyles_unstable;
    },
    useBreadcrumbButton_unstable: function() {
        return _BreadcrumbButton.useBreadcrumbButton_unstable;
    },
    useBreadcrumbContext_unstable: function() {
        return _Breadcrumb.useBreadcrumbContext_unstable;
    },
    useBreadcrumbDividerStyles_unstable: function() {
        return _BreadcrumbDivider.useBreadcrumbDividerStyles_unstable;
    },
    useBreadcrumbDivider_unstable: function() {
        return _BreadcrumbDivider.useBreadcrumbDivider_unstable;
    },
    useBreadcrumbItemStyles_unstable: function() {
        return _BreadcrumbItem.useBreadcrumbItemStyles_unstable;
    },
    useBreadcrumbItem_unstable: function() {
        return _BreadcrumbItem.useBreadcrumbItem_unstable;
    },
    useBreadcrumbStyles_unstable: function() {
        return _Breadcrumb.useBreadcrumbStyles_unstable;
    },
    useBreadcrumb_unstable: function() {
        return _Breadcrumb.useBreadcrumb_unstable;
    }
});
const _Breadcrumb = require("./Breadcrumb");
const _BreadcrumbDivider = require("./BreadcrumbDivider");
const _BreadcrumbItem = require("./BreadcrumbItem");
const _index = require("./utils/index");
const _BreadcrumbButton = require("./BreadcrumbButton");
