// @fluentui/react-card/lib-commonjs/components/Card/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Card: function() {
        return _Card.Card;
    },
    CardProvider: function() {
        return _CardContext.CardProvider;
    },
    cardCSSVars: function() {
        return _useCardStylesstyles.cardCSSVars;
    },
    cardClassNames: function() {
        return _useCardStylesstyles.cardClassNames;
    },
    cardContextDefaultValue: function() {
        return _CardContext.cardContextDefaultValue;
    },
    renderCard_unstable: function() {
        return _renderCard.renderCard_unstable;
    },
    useCardContext_unstable: function() {
        return _CardContext.useCardContext_unstable;
    },
    useCardStyles_unstable: function() {
        return _useCardStylesstyles.useCardStyles_unstable;
    },
    useCard_unstable: function() {
        return _useCard.useCard_unstable;
    }
});
const _Card = require("./Card");
const _CardContext = require("./CardContext");
const _renderCard = require("./renderCard");
const _useCard = require("./useCard");
const _useCardStylesstyles = require("./useCardStyles.styles");
