// @fluentui/react-avatar/lib-commonjs/components/AvatarGroupPopover/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AvatarGroupPopover: function() {
        return _AvatarGroupPopover.AvatarGroupPopover;
    },
    avatarGroupPopoverClassNames: function() {
        return _useAvatarGroupPopoverStylesstyles.avatarGroupPopoverClassNames;
    },
    renderAvatarGroupPopover_unstable: function() {
        return _renderAvatarGroupPopover.renderAvatarGroupPopover_unstable;
    },
    useAvatarGroupPopoverContextValues_unstable: function() {
        return _useAvatarGroupPopoverContextValues.useAvatarGroupPopoverContextValues_unstable;
    },
    useAvatarGroupPopoverStyles_unstable: function() {
        return _useAvatarGroupPopoverStylesstyles.useAvatarGroupPopoverStyles_unstable;
    },
    useAvatarGroupPopover_unstable: function() {
        return _useAvatarGroupPopover.useAvatarGroupPopover_unstable;
    }
});
const _AvatarGroupPopover = require("./AvatarGroupPopover");
const _renderAvatarGroupPopover = require("./renderAvatarGroupPopover");
const _useAvatarGroupPopover = require("./useAvatarGroupPopover");
const _useAvatarGroupPopoverStylesstyles = require("./useAvatarGroupPopoverStyles.styles");
const _useAvatarGroupPopoverContextValues = require("./useAvatarGroupPopoverContextValues");
