// @fluentui/react-provider/lib-commonjs/FluentProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FluentProvider: function() {
        return _index.FluentProvider;
    },
    createCSSRuleFromTheme: function() {
        return _index.createCSSRuleFromTheme;
    },
    fluentProviderClassNames: function() {
        return _index.fluentProviderClassNames;
    },
    renderFluentProvider_unstable: function() {
        return _index.renderFluentProvider_unstable;
    },
    useFluentProviderContextValues_unstable: function() {
        return _index.useFluentProviderContextValues_unstable;
    },
    useFluentProviderStyles_unstable: function() {
        return _index.useFluentProviderStyles_unstable;
    },
    useFluentProviderThemeStyleTag: function() {
        return _index.useFluentProviderThemeStyleTag;
    },
    useFluentProvider_unstable: function() {
        return _index.useFluentProvider_unstable;
    }
});
const _index = require("./components/FluentProvider/index");
