// @fluentui/react-table/lib-commonjs/components/TableBody/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TableBody: function() {
        return _TableBody.TableBody;
    },
    renderTableBody_unstable: function() {
        return _renderTableBody.renderTableBody_unstable;
    },
    tableBodyClassName: function() {
        return _useTableBodyStylesstyles.tableBodyClassName;
    },
    tableBodyClassNames: function() {
        return _useTableBodyStylesstyles.tableBodyClassNames;
    },
    useTableBodyStyles_unstable: function() {
        return _useTableBodyStylesstyles.useTableBodyStyles_unstable;
    },
    useTableBody_unstable: function() {
        return _useTableBody.useTableBody_unstable;
    }
});
const _TableBody = require("./TableBody");
const _renderTableBody = require("./renderTableBody");
const _useTableBody = require("./useTableBody");
const _useTableBodyStylesstyles = require("./useTableBodyStyles.styles");
