// @fluentui/react-menu/lib-commonjs/components/MenuGroup/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MenuGroup: function() {
        return _MenuGroup.MenuGroup;
    },
    menuGroupClassNames: function() {
        return _useMenuGroupStylesstyles.menuGroupClassNames;
    },
    renderMenuGroup_unstable: function() {
        return _renderMenuGroup.renderMenuGroup_unstable;
    },
    useMenuGroupContextValues_unstable: function() {
        return _useMenuGroupContextValues.useMenuGroupContextValues_unstable;
    },
    useMenuGroupStyles_unstable: function() {
        return _useMenuGroupStylesstyles.useMenuGroupStyles_unstable;
    },
    useMenuGroup_unstable: function() {
        return _useMenuGroup.useMenuGroup_unstable;
    }
});
const _MenuGroup = require("./MenuGroup");
const _renderMenuGroup = require("./renderMenuGroup");
const _useMenuGroup = require("./useMenuGroup");
const _useMenuGroupContextValues = require("./useMenuGroupContextValues");
const _useMenuGroupStylesstyles = require("./useMenuGroupStyles.styles");
