// @fluentui/react-accordion/lib-commonjs/components/Accordion/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Accordion: function() {
        return _Accordion.Accordion;
    },
    accordionClassNames: function() {
        return _useAccordionStylesstyles.accordionClassNames;
    },
    renderAccordion_unstable: function() {
        return _renderAccordion.renderAccordion_unstable;
    },
    useAccordionContextValues_unstable: function() {
        return _useAccordionContextValues.useAccordionContextValues_unstable;
    },
    useAccordionStyles_unstable: function() {
        return _useAccordionStylesstyles.useAccordionStyles_unstable;
    },
    useAccordion_unstable: function() {
        return _useAccordion.useAccordion_unstable;
    }
});
const _Accordion = require("./Accordion");
const _renderAccordion = require("./renderAccordion");
const _useAccordion = require("./useAccordion");
const _useAccordionStylesstyles = require("./useAccordionStyles.styles");
const _useAccordionContextValues = require("./useAccordionContextValues");
