// @fluentui/react-menu/lib-commonjs/components/MenuItemSwitch/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MenuItemSwitch: function() {
        return _MenuItemSwitch.MenuItemSwitch;
    },
    circleFilledClassName: function() {
        return _useMenuItemSwitchStylesstyles.circleFilledClassName;
    },
    menuItemSwitchClassNames: function() {
        return _useMenuItemSwitchStylesstyles.menuItemSwitchClassNames;
    },
    renderMenuItemSwitch_unstable: function() {
        return _renderMenuItemSwitch.renderMenuItemSwitch_unstable;
    },
    useMenuItemSwitchStyles_unstable: function() {
        return _useMenuItemSwitchStylesstyles.useMenuItemSwitchStyles_unstable;
    },
    useMenuItemSwitch_unstable: function() {
        return _useMenuItemSwitch.useMenuItemSwitch_unstable;
    }
});
const _MenuItemSwitch = require("./MenuItemSwitch");
const _renderMenuItemSwitch = require("./renderMenuItemSwitch");
const _useMenuItemSwitch = require("./useMenuItemSwitch");
const _useMenuItemSwitchStylesstyles = require("./useMenuItemSwitchStyles.styles");
