// @fluentui/react-link/lib-commonjs/Link.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Link: function() {
        return _index.Link;
    },
    LinkContextProvider: function() {
        return _contexts.LinkContextProvider;
    },
    linkClassNames: function() {
        return _index.linkClassNames;
    },
    linkContextDefaultValue: function() {
        return _contexts.linkContextDefaultValue;
    },
    renderLink_unstable: function() {
        return _index.renderLink_unstable;
    },
    useLinkContext: function() {
        return _contexts.useLinkContext;
    },
    useLinkState_unstable: function() {
        return _index.useLinkState_unstable;
    },
    useLinkStyles_unstable: function() {
        return _index.useLinkStyles_unstable;
    },
    useLink_unstable: function() {
        return _index.useLink_unstable;
    }
});
const _index = require("./components/Link/index");
const _contexts = require("./contexts");
