// @fluentui/react-motion-components-preview/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Collapse: function() {
        return _Collapse.Collapse;
    },
    CollapseDelayed: function() {
        return _Collapse.CollapseDelayed;
    },
    CollapseRelaxed: function() {
        return _Collapse.CollapseRelaxed;
    },
    CollapseSnappy: function() {
        return _Collapse.CollapseSnappy;
    },
    Fade: function() {
        return _Fade.Fade;
    },
    FadeRelaxed: function() {
        return _Fade.FadeRelaxed;
    },
    FadeSnappy: function() {
        return _Fade.FadeSnappy;
    },
    Scale: function() {
        return _Scale.Scale;
    },
    ScaleRelaxed: function() {
        return _Scale.ScaleRelaxed;
    },
    ScaleSnappy: function() {
        return _Scale.ScaleSnappy;
    },
    createCollapseDelayedPresence: function() {
        return _Collapse.createCollapseDelayedPresence;
    },
    createCollapsePresence: function() {
        return _Collapse.createCollapsePresence;
    },
    createFadePresence: function() {
        return _Fade.createFadePresence;
    },
    createScalePresence: function() {
        return _Scale.createScalePresence;
    }
});
const _Collapse = require("./components/Collapse");
const _Fade = require("./components/Fade");
const _Scale = require("./components/Scale");
