// @fluentui/react-rating/lib-commonjs/components/Rating/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Rating: function() {
        return _Rating.Rating;
    },
    ratingClassNames: function() {
        return _useRatingStylesstyles.ratingClassNames;
    },
    renderRating_unstable: function() {
        return _renderRating.renderRating_unstable;
    },
    useRatingContextValues: function() {
        return _useRatingContextValues.useRatingContextValues;
    },
    useRatingStyles_unstable: function() {
        return _useRatingStylesstyles.useRatingStyles_unstable;
    },
    useRating_unstable: function() {
        return _useRating.useRating_unstable;
    }
});
const _Rating = require("./Rating");
const _renderRating = require("./renderRating");
const _useRating = require("./useRating");
const _useRatingStylesstyles = require("./useRatingStyles.styles");
const _useRatingContextValues = require("./useRatingContextValues");
