// @fluentui/react-virtualizer/lib-commonjs/components/Virtualizer/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Virtualizer: function() {
        return _Virtualizer.Virtualizer;
    },
    renderVirtualizerChildPlaceholder: function() {
        return _renderVirtualizer.renderVirtualizerChildPlaceholder;
    },
    renderVirtualizer_unstable: function() {
        return _renderVirtualizer.renderVirtualizer_unstable;
    },
    useVirtualizerStyles_unstable: function() {
        return _useVirtualizerStylesstyles.useVirtualizerStyles_unstable;
    },
    useVirtualizer_unstable: function() {
        return _useVirtualizer.useVirtualizer_unstable;
    },
    virtualizerClassNames: function() {
        return _useVirtualizerStylesstyles.virtualizerClassNames;
    }
});
const _Virtualizer = require("./Virtualizer");
const _useVirtualizer = require("./useVirtualizer");
const _renderVirtualizer = require("./renderVirtualizer");
const _useVirtualizerStylesstyles = require("./useVirtualizerStyles.styles");
