// @fluentui/react-color-picker/lib-commonjs/components/AlphaSlider/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AlphaSlider: function() {
        return _AlphaSlider.AlphaSlider;
    },
    alphaSliderCSSVars: function() {
        return _useAlphaSliderStylesstyles.alphaSliderCSSVars;
    },
    alphaSliderClassNames: function() {
        return _useAlphaSliderStylesstyles.alphaSliderClassNames;
    },
    renderAlphaSlider_unstable: function() {
        return _renderAlphaSlider.renderAlphaSlider_unstable;
    },
    useAlphaSliderStyles_unstable: function() {
        return _useAlphaSliderStylesstyles.useAlphaSliderStyles_unstable;
    },
    useAlphaSlider_unstable: function() {
        return _useAlphaSlider.useAlphaSlider_unstable;
    }
});
const _AlphaSlider = require("./AlphaSlider");
const _renderAlphaSlider = require("./renderAlphaSlider");
const _useAlphaSlider = require("./useAlphaSlider");
const _useAlphaSliderStylesstyles = require("./useAlphaSliderStyles.styles");
