// @fluentui/react-table/lib-commonjs/components/TableHeaderCell/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TableHeaderCell: function() {
        return _TableHeaderCell.TableHeaderCell;
    },
    renderTableHeaderCell_unstable: function() {
        return _renderTableHeaderCell.renderTableHeaderCell_unstable;
    },
    tableHeaderCellClassName: function() {
        return _useTableHeaderCellStylesstyles.tableHeaderCellClassName;
    },
    tableHeaderCellClassNames: function() {
        return _useTableHeaderCellStylesstyles.tableHeaderCellClassNames;
    },
    useTableHeaderCellStyles_unstable: function() {
        return _useTableHeaderCellStylesstyles.useTableHeaderCellStyles_unstable;
    },
    useTableHeaderCell_unstable: function() {
        return _useTableHeaderCell.useTableHeaderCell_unstable;
    }
});
const _TableHeaderCell = require("./TableHeaderCell");
const _renderTableHeaderCell = require("./renderTableHeaderCell");
const _useTableHeaderCell = require("./useTableHeaderCell");
const _useTableHeaderCellStylesstyles = require("./useTableHeaderCellStyles.styles");
