// @fluentui/react-teaching-popover/lib-commonjs/components/TeachingPopoverBody/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TeachingPopoverBody: function() {
        return _TeachingPopoverBody.TeachingPopoverBody;
    },
    renderTeachingPopoverBody_unstable: function() {
        return _renderTeachingPopoverBody.renderTeachingPopoverBody_unstable;
    },
    teachingPopoverBodyClassNames: function() {
        return _useTeachingPopoverBodyStylesstyles.teachingPopoverBodyClassNames;
    },
    useMediaStyles: function() {
        return _useTeachingPopoverBodyStylesstyles.useMediaStyles;
    },
    useTeachingPopoverBodyStyles_unstable: function() {
        return _useTeachingPopoverBodyStylesstyles.useTeachingPopoverBodyStyles_unstable;
    },
    useTeachingPopoverBody_unstable: function() {
        return _useTeachingPopoverBody.useTeachingPopoverBody_unstable;
    }
});
const _TeachingPopoverBody = require("./TeachingPopoverBody");
const _renderTeachingPopoverBody = require("./renderTeachingPopoverBody");
const _useTeachingPopoverBody = require("./useTeachingPopoverBody");
const _useTeachingPopoverBodyStylesstyles = require("./useTeachingPopoverBodyStyles.styles");
