// @fluentui/react-avatar/lib-commonjs/components/AvatarGroupItem/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AvatarGroupItem: function() {
        return _AvatarGroupItem.AvatarGroupItem;
    },
    avatarGroupItemClassNames: function() {
        return _useAvatarGroupItemStylesstyles.avatarGroupItemClassNames;
    },
    renderAvatarGroupItem_unstable: function() {
        return _renderAvatarGroupItem.renderAvatarGroupItem_unstable;
    },
    useAvatarGroupItemStyles_unstable: function() {
        return _useAvatarGroupItemStylesstyles.useAvatarGroupItemStyles_unstable;
    },
    useAvatarGroupItem_unstable: function() {
        return _useAvatarGroupItem.useAvatarGroupItem_unstable;
    },
    useGroupChildClassName: function() {
        return _useAvatarGroupItemStylesstyles.useGroupChildClassName;
    }
});
const _AvatarGroupItem = require("./AvatarGroupItem");
const _renderAvatarGroupItem = require("./renderAvatarGroupItem");
const _useAvatarGroupItem = require("./useAvatarGroupItem");
const _useAvatarGroupItemStylesstyles = require("./useAvatarGroupItemStyles.styles");
