// @fluentui/react-teaching-popover/lib-commonjs/components/TeachingPopoverCarousel/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TeachingPopoverCarousel: function() {
        return _TeachingPopoverCarousel.TeachingPopoverCarousel;
    },
    renderTeachingPopoverCarousel_unstable: function() {
        return _renderTeachingPopoverCarousel.renderTeachingPopoverCarousel_unstable;
    },
    teachingPopoverCarouselClassNames: function() {
        return _useTeachingPopoverCarouselStylesstyles.teachingPopoverCarouselClassNames;
    },
    useTeachingPopoverCarouselContextValues_unstable: function() {
        return _useTeachingPopoverCarouselContextValues.useTeachingPopoverCarouselContextValues_unstable;
    },
    useTeachingPopoverCarouselStyles_unstable: function() {
        return _useTeachingPopoverCarouselStylesstyles.useTeachingPopoverCarouselStyles_unstable;
    },
    useTeachingPopoverCarousel_unstable: function() {
        return _useTeachingPopoverCarousel.useTeachingPopoverCarousel_unstable;
    }
});
const _TeachingPopoverCarousel = require("./TeachingPopoverCarousel");
const _renderTeachingPopoverCarousel = require("./renderTeachingPopoverCarousel");
const _useTeachingPopoverCarousel = require("./useTeachingPopoverCarousel");
const _useTeachingPopoverCarouselStylesstyles = require("./useTeachingPopoverCarouselStyles.styles");
const _useTeachingPopoverCarouselContextValues = require("./useTeachingPopoverCarouselContextValues");
